### PR TITLE
Migrate to ProtoBuf.jl 1.0

### PR DESCRIPTION
**Pull request by Drvi**
Recently, we published a new version of ProtoBuf.jl ([discourse link](https://discourse.julialang.org/t/ann-protobuf-jl-1-0-0/85885)), and this PR migrates to the new version.

Sadly, the new version doesn't (yet) support comment/docstring preservation, hopefully, it is ok to lose them for now.

### CODE REVIEW

**Comment by NHDaly** [comment]
Thanks @Drvi! The old version didn't support it either, since my PR was never merged: https://github.com/JuliaIO/ProtoBuf.jl/pull/128 😢 

But we generated these from that PR. Anyway, yeah, totally fine to lose them 👍 Hopefully that can be added back at some point though, because the pprof proto spec is MAD confusing! 😅 

Thanks :)

File: src/flamegraphs.jl [hunk 1]
```diff
@@ -82,26 +80,17 @@ function pprof(fg::Node{NodeData},
...
-
-    if drop_frames !== nothing
-        prof.drop_frames = enter!(drop_frames)
-    end
-    if keep_frames !== nothing
-        prof.keep_frames = enter!(keep_frames)
-    end
+    period_type = ValueType!("cpu", "nanoseconds")
+    drop_frames = isnothing(drop_frames) ? 0 : enter!(drop_frames)
+    keep_frames = isnothing(keep_frames) ? 0 : enter!(keep_frames)
```
Comment (NHDaly) [inline src/flamegraphs.jl:92]: Is 0 the correct default value for these when there is no provided value? 🤔

**Review by NHDaly — APPROVED** [review]
❤️ amazing. Thanks @Drvi! 🚀  LGTM.

Someone should manually test this out a bit to make sure it looks okay, but LGTM

File: src/flamegraphs.jl (hunk 1 above)
Comment (Drvi) [reply src/flamegraphs.jl:92]: Yeah, so when default value is not changed in the proto file via options, 0 is the default for numeric scalars

**Comment by NHDaly** [comment]
@Drvi: It looks like ProtoBuf.jl 1.0 doesn't support julia 1.3 anymore?

I think we can probably bump up our minimum supported julia version with this release. That seems reasonable to me. 👍

**Comment by Drvi** [comment]
Thank you, @NHDaly! I tried locally on `peakflops` and seemed fine to me, but would be great if someone else could verify.

I just noticed that the julia 1.3 tests are failing because ProtoBuf targets 1.7+... is it ok to bump the minimum supported version here as well?

**Review by NHDaly — APPROVED** [review]
Excellent, thank you for fixing the julia version!

**Comment by NHDaly** [comment]
🤔 is it possible to keep our minimum supported julia version to be 1.6, since that's the LTS? Profiling is a fairly important thing, and I'd _like_ to keep compatibility with 1.6+ if possible.. Is ProtoBuf.jl only 1.7+? Does it need to be 1.7+?

**Comment by Drvi** [comment]
That is a really good point, I think we can support 1.6 -- trying here https://github.com/JuliaIO/ProtoBuf.jl/pull/203

**Comment by Drvi** [comment]
Locally, all tests were passing for me on 1.6

**Comment by Drvi** [comment]
Ok, v1.0.1 of ProtoBuf should be in the general registry soon and with it, we should be able to support 1.6 here.

**Comment by Drvi** [comment]
Please feel free to merge if you are happy with the changes 👍

**Comment by NHDaly** [comment]
amazing. Thanks for the quick turnaround on 1.6 support! ❤️ and thanks for the PR!

**Pull request merged**